src/spikeinterface/preprocessing/decimate.py

```diff
@@ -123,13 +123,6 @@ def get_num_samples(self):
         return int(np.ceil((parent_n_samp - self._decimation_offset) / self._decimation_factor))
 
     def get_traces(self, start_frame, end_frame, channel_indices):
-        if start_frame is None:
-            start_frame = 0
-        if end_frame is None:
-            end_frame = self.get_num_samples()
-        end_frame = min(end_frame, self.get_num_samples())
-        start_frame = min(start_frame, self.get_num_samples())
```

Comment:
This specific last lines is strange but this was put here for a reason a guess.
Here this (useless) protection has gone.

Reply:
I agree it was strange, but I don't think there was a reason. We can improve this global behavior later to follow #1989